Assets/SEE/Tools/LSP/LSPHandler.cs

```diff
@@ -0,0 +1,627 @@
+            finally
+            {
+                semaphore.Release();
+                spinner.Dispose();
+            }
+            return;
+
+            void HandleInitialWorkDoneProgress(WorkDoneProgressCreateParams progressParams)
+            {
+                if (!IsReady && progressParams.Token != null)
+                {
+                    initialWork.Add(progressParams.Token);
+                    MonitorInitialWorkDoneProgress(progressParams.Token).Forget();
+                }
+            }
+
+            async UniTaskVoid MonitorInitialWorkDoneProgress(ProgressToken token)
+            {
+                await foreach (WorkDoneProgress _ in Client.WorkDoneManager.Monitor(token).ToUniTaskAsyncEnumerable()
+                                                           .Where(x => x.Kind == WorkDoneProgressKind.End))
+                {
+                    initialWork.Remove(token);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Handles the ShowMessage notification with the given <paramref name="messageParams"/>
+        /// by showing a notification to the user.
+        /// </summary>
+        /// <param name="showMessageParams">The parameters of the ShowMessage notification.</param>
+        private void ShowMessage(ShowMessageParams showMessageParams)
+        {
+            string languageServerName = Server?.Name ?? "Language Server";
+            switch (showMessageParams.Type)
+            {
+                case MessageType.Error:
+                    ShowNotification.Error($"{languageServerName} Error", showMessageParams.Message);
+                    break;
+                case MessageType.Warning:
+                    ShowNotification.Warn($"{languageServerName} Warning", showMessageParams.Message);
+                    break;
+                case MessageType.Info:
+                    ShowNotification.Info($"{languageServerName} Info", showMessageParams.Message);
+                    break;
+                case MessageType.Log:
+                default:
+                    ShowNotification.Info($"{languageServerName} Log", showMessageParams.Message);
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Handles the LogMessage notification with the given <paramref name="messageParams"/>
+        /// by logging the message to the Unity console.
+        /// </summary>
+        /// <param name="messageParams">The parameters of the LogMessage notification.</param>
+        private static void LogMessage(LogMessageParams messageParams)
+        {
+            switch (messageParams.Type)
+            {
+                case MessageType.Error:
+                    Debug.LogError(messageParams.Message);
+                    break;
+                case MessageType.Warning:
+                    Debug.LogWarning(messageParams.Message);
+                    break;
+                case MessageType.Info:
+                case MessageType.Log:
```

Comment:
You can use an `or` between two cases: `case MessageType.Info or MessageType.Log:`.
That is more readable.

Reply:
I fixed it.